testapp/src/main/java/com/rakuten/tech/mobile/testapp/analytics/rat_wrapper/RATActivity.kt

```diff
@@ -0,0 +1,118 @@
+package com.rakuten.tech.mobile.testapp.analytics.rat_wrapper
+
+import android.view.MenuItem
+import androidx.appcompat.app.ActionBar
+import androidx.appcompat.app.AppCompatActivity
+import com.rakuten.tech.mobile.testapp.analytics.DemoAppAnalytics
+import com.rakuten.tech.mobile.testapp.ui.settings.AppSettings
+
+enum class MenuItemDefaults() {
+    HOME,
+    BACK,
+}
+
+open class RATActivity : AppCompatActivity(), IRatActivity {
+    private var ratEvent: RATEvent? = null
+    private var ratEventMenuItem: RATEvent? = null
+    private var menu_item_label = ""
+    private var screen_name = ""
+
+    /** Will send Appear event when open a activity. */
+    override fun onResume() {
+        sendAnalyticsDefault()
+        super.onResume()
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (item.itemId == android.R.id.home) {
+            supportActionBar?.let {
+                menu_item_label = if (it.displayOptions and ActionBar.DISPLAY_HOME_AS_UP !== 0) {
+                    MenuItemDefaults.BACK.name
+                } else {
+                    MenuItemDefaults.HOME.name
+                }
+            }
+        } else {
+            menu_item_label = item.title.toString()
+        }
+        sendAnalyticsForMenuItem()
+        return super.onOptionsItemSelected(item)
+    }
+
+    /** Will send click event when any of the menu item is being clicked. */
+    private fun sendAnalyticsForMenuItem() {
+        prepareMenuItemClickEventToSend()
+        ratEventMenuItem?.let {
+            DemoAppAnalytics.init(AppSettings.instance.projectId).sendAnalytics(it)
+        }
+    }
+
+    /** Will send Appear event when open a activity. */
+    private fun sendAnalyticsDefault() {
+        prepareScreenAppearEventToSend()
+        ratEvent?.let {
+            DemoAppAnalytics.init(AppSettings.instance.projectId).sendAnalytics(it)
+        }
+    }
+
+    /** Prepare default event if custom RAT event is not set. */
+    override fun prepareMenuItemClickEventToSend() {
+        if (ratEventMenuItem == null)
+            ratEventMenuItem = RATEvent(
+                event = EventType.CLICK,
+                action = ActionType.OPEN,
+                label = menu_item_label,
+                screen_name = getScreenName()
+            )
+    }
+
+    /** Prepare default event if custom RAT event is not set. */
+    override fun prepareScreenAppearEventToSend() {
+        if (ratEvent == null)
+            ratEvent = RATEvent(
+                event = EventType.APPEAR,
+                action = ActionType.OPEN,
```

Comment:
This event should be `EventType.PAGE_LOAD`

The `appear` event is supposed to be used for clickable page elements such as buttons - that event is sent when the item appears.